### PR TITLE
Cleanup the assembly AST

### PR DESCRIPTION
**Pull request by axic**
Rename Assignment to StackAssignment and FunctionalAssignment to Assignment as those make more sense.

### CODE REVIEW

File: libsolidity/inlineasm/AsmParser.cpp [hunk 1]
```diff
@@ -401,3 +393,10 @@ string Parser::expectAsmIdentifier()
 	expectToken(Token::Identifier);
 	return name;
 }
+
+void Parser::expectSinglePusher(dev::solidity::Instruction const& _instruction)
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmParser.cpp]: Is `dup1` a valid single pusher or not?

Comment (axic) [reply libsolidity/inlineasm/AsmParser.cpp]: DUP1 has `ret = 2`, so that should be fine

Comment (chriseth) [reply libsolidity/inlineasm/AsmParser.cpp]: ah right, ok!

**Review by chriseth — APPROVED** [review]

File: libsolidity/inlineasm/AsmParser.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/inlineasm/AsmParser.cpp]: It does fail on on zero pushers (jump, revert, etc) so removed it from this PR, we can address it separately.

**Pull request merged**